nilearn/signal.py

```diff
@@ -48,12 +49,12 @@ def _standardize(signals, detrend=False, normalize=True):
     if signals.shape[0] == 1:
         warnings.warn('Standardization of 3D signal has been requested but '
             'would lead to zero values. Skipping.')
-        return signals
+        return signals.copy()
```

Comment:
I got an headache trying to find out if a copy is done in all the call tree. I would either:

* do an explicit copy at the beginning of the function, and the rest inplace
* do everything inplace and make an explicit copy above

For the moment, there is a bug since the code expects _standardize to do a copy of signals and the copy is not done if no treratment is requested.